### PR TITLE
✨ Add controller tests for HelmReleaseProxy

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Jont828** [comment]
@CecileRobertMichon Any thoughts?

File: controllers/helmreleaseproxy/helmreleaseproxy_controller.go [hunk 1]
```diff
@@ -221,6 +223,7 @@ func (r *HelmReleaseProxyReconciler) reconcileNormal(ctx context.Context, helmRe
 		if status == helmRelease.StatusDeployed {
 			conditions.MarkTrue(helmReleaseProxy, addonsv1alpha1.HelmReleaseReadyCondition)
 		} else if status.IsPending() {
+			fmt.Printf("Is pending: %s\n", status)
```
Comment (CecileRobertMichon) [inline controllers/helmreleaseproxy/helmreleaseproxy_controller.go:226]: leftover debug print?

File: internal/helm_client.go [hunk 2]
```diff
@@ -45,6 +45,16 @@ import (
 	addonsv1alpha1 "sigs.k8s.io/cluster-api-addon-provider-helm/api/v1alpha1"
 )
 
+type Client interface {
+	InstallOrUpgradeHelmRelease(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec) (*helmRelease.Release, error)
+	// InstallHelmRelease(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec) (*helmRelease.Release, error)
```
Comment (CecileRobertMichon) [inline internal/helm_client.go:50]: why are these commented out?

File: internal/helm_client.go [hunk 3]
```diff
@@ -45,6 +45,16 @@ import (
...
+	// UpgradeHelmReleaseIfChanged(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec, existing *helmRelease.Release) (*helmRelease.Release, error)
+	GetHelmRelease(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec) (*helmRelease.Release, error)
+	UninstallHelmRelease(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec) (*helmRelease.UninstallReleaseResponse, error)
+}
+
+type HelmClient struct{}
```
Comment (CecileRobertMichon) [inline internal/helm_client.go:56]: why is this an empty struct?

File: internal/kubeconfig.go [hunk 4]
```diff
@@ -22,36 +22,68 @@ import (
...
+	}
+
+	c, err := client.New("")
+	if err != nil {
+		return "", err
+	}
+
+	options := client.GetKubeconfigOptions{
+		Kubeconfig: client.Kubeconfig(*managementKubeconfig),
+		// Kubeconfig:          client.Kubeconfig{Path: gk.kubeconfig, Context: gk.kubeconfigContext},
```
Comment (CecileRobertMichon) [inline internal/kubeconfig.go:58]: commented out code

File: internal/helm_client.go (hunk 3 above)
Comment (Jont828) [reply internal/helm_client.go:56]: We need a struct that implements the `Client` interface above in order to perform the calls. It's similar to how CAPZ is creating `AzureClient` structs that implement `CreateOrUpdate()`. We're using the struct below to set up functions like this:

```
func (c *HelmClient) InstallOrUpgradeHelmRelease(ctx context.Context, kubeconfig string, spec addonsv1alpha1.HelmReleaseProxySpec) (*helmRelease.Release, error)
```

File: internal/helm_client.go (hunk 2 above)
Comment (Jont828) [reply internal/helm_client.go:50]: Forgot to delete!

**Comment by Jont828** [comment]
@CecileRobertMichon Do you think we're good to go on this? I'm hoping to merge this and cut another release.

**Comment by CecileRobertMichon** [comment]
want to squash?

lgtm

**Comment by Jont828** [comment]
Done, just going to wait for tests.

**Comment by Jont828** [comment]
@CecileRobertMichon Good to go!

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**